File: Two-Sum.cs
1public class Solution {
2    public int[] TwoSum(int[] nums, int target) {
3        Dictionary<int,int> map = new Dictionary<int,int>();
4        
5        for(var i = 0 ; i < nums.Length ; i++){
6            var diff = target - nums[i];
7            if(map.ContainsKey(diff)){
8                return new int[]{i, map[diff] };
9            }
10
11            if(!map.ContainsKey(nums[i])){
12                map.Add(nums[i], i);
13            }
14        }
15        return new int[0];
16    }
17}

File: Two-Sum.py
1class Solution:
2    def twoSum(self, nums: List[int], target: int) -> List[int]:
3        seen = {}
4        for i, num in enumerate(nums):
5            sum = target - num
6            if sum in seen:
7                return [seen[sum], i]
8            seen[num] = i